### DISCUSSION TITLE
Replacing the use of os.path.abspath in unittest

### DISCUSSION THREAD

**Discussion by Pablompg** [manusimidt/py-xbrl#16] [Ideas]
I believe it would be better to use relative paths in testing instead of absolute paths. The absolute path depends not on the file itself, but from where you are running it.

The usual way of running tests is from the root directory of the proyect, and because of using absolute paths in os.path.abspath it fails. What do you think about changing it?

**Comment by manusimidt** [comment]
Yes, correct, i also think that using relative paths is better, expecially because i use relative paths everywhere else (i.e: in Readme.md).

I have switched to relative paths for the unit tests of the instance, taxonomy and linkbase modules in https://github.com/manusimidt/xbrl_parser/commit/d8459097f25ea9f563e880f20b41886586535d3e.

I still have to put in some work into the test_uri_resolver unittest. Initially i struggled a lot because i wanted to create one fancy method that resolved both relative url's and paths. It probably would have been much easier if i had just created one function for url's and one for file paths... 
Then I would have been able to use functions like `os.path.join()` to resolve relative paths and i wouldn't have to deal with different directory separators.

**Reply by manusimidt** [reply]
The issue with relative paths for the xbrl parser is the following:

say we have the following xlink of a linkbase in a taxonomy schema:
```xml
<link:linkbaseRef [..] xlink:href="./../example_lab.xml" [..]/>
```
To import the linkbase `example_lab.xml` we have to distinguish between two cases:

**1. The schema that we are trying to parse is stored locally:**
Then the linkbase will also be imported from the disk and not fetched from the internet:

**2. The schema that we are trying to parse is stored remotely:**
Then the linkbase will also be fetched from the internet


This is handled by the following code:
https://github.com/manusimidt/xbrl_parser/blob/5a1b3882e8b65f7815bbc330c9c010169c195951/xbrl_parser/taxonomy.py#L286-L296


A little more simplified:

1. If the taxonomy_uri is `E:\\data\\xbrl\\us-gaap\\2021\\elts\\us-gaap-2021-01-31.xsd` the parser will get the linkbase from `E:\\data\\xbrl\\us-gaap\\2021\\example_lab.xml`

2. If the taxonomy_uri is `https://xbrl.fasb.org/us-gaap/2021/elts/us-gaap-2021-01-31.xsd` the parser will fetch the linkbase from `https://xbrl.fasb.org/us-gaap/2021/example_lab.xml`

**Reply by Pablompg** [reply]
Great job changing the absolute paths to relative in unittests. Thank you !
